lua/bufferline/state.lua

```diff
@@ -77,22 +77,18 @@ end
 
 --- Get the list of buffers
 function state.get_buffer_list()
-  local buffers = list_bufs()
   local result = {}
 
   local exclude_ft = options.exclude_ft()
   local exclude_name = options.exclude_name()
   local hide_extensions = options.hide().extensions
 
-  for _, buffer in ipairs(buffers) do
-    if buf_get_option(buffer, 'buflisted') then
-      local ft = buf_get_option(buffer, 'filetype')
-      if not utils.has(exclude_ft, ft) then
-        local name = utils.basename(buf_get_name(buffer), hide_extensions)
-        if not utils.has(exclude_name, name) then
-          result[#result + 1] = buffer
-        end
-      end
+  for _, bufnr in ipairs(list_bufs()) do
```

Comment:
I'm willing to revert this. It was leftover cleanup from when I was implementing `hide.unlisted`, so I figured I'd keep it— since it's all one `if`, maybe LuaJIT can do more optimizations with it? Who knows.

Reply:
one `if`? not sure what you mean here.

> maybe LuaJIT can do more optimizations with it

Terser code doesn't result in more performant code.

Reply:
Terser code doesn't produce more efficient bytecode, I would revert this.